src/Workspaces/Remote/ServiceHub/Shared/ClientDirectStream.cs

```diff
@@ -43,7 +43,9 @@ private class ClientDirectStream : Stream
             private static readonly TimeSpan ConnectRetryIntervalMs = TimeSpan.FromMilliseconds(20);
 
             private readonly string _name;
+#pragma warning disable CA2213 // Disposable fields should be disposed - _stream wraps _pipe and is disposed in Close.
```

Comment:
💭 The code would be clearer to machines if `_pipe` was wrapped in [`DelegatingStream`](https://github.com/sharwell/openstack.net/blob/master/src/OpenStack.Net/OpenStack/IO/DelegatingStream.cs) when passed to `_stream`, allowing both of these fields to be disposed in `ClientDirectStream.Dispose`.

I'm not not confident the pattern of overriding `Close()` and `Dispose(bool)` is correct, and it certainly isn't clear. It would be preferable to remove the override of `Close()` and place all of the cleanup logic in `Dispose(bool)` (which is already called by `Close()` in the base class).